neural_compressor/torch/quantization/quantize.py

```diff
@@ -59,11 +61,19 @@ def quantize(
     logger.info(quant_config.to_dict())
     # select quantization algo according to config
 
-    model_info = quant_config.get_model_info(model=q_model)
+    if is_ipex_available:
```

Comment:
Consider unifying the `get_model_info`  API as the suggestion for `algo_func`. If possible, let's avoid introducing too many if-else.